salt/client/__init__.py

```diff
@@ -248,7 +248,7 @@ def _check_pub_data(self, pub_data):
         '''
         Common checks on the pub_data data structure returned from running pub
         '''
-        if not pub_data:
+        if pub_data == '':
```

Comment:
@cachedout I am curious what is the difference: is it like `assert pub_data is None` is OK and we don't need to raise this exception?